apollo-federation/src/supergraph/mod.rs

```diff
@@ -672,25 +684,52 @@ fn add_empty_type(
                     }
                     .into());
                 }
-                TypeDefinitionPosition::Object(pos) => {
```

Comment:
Refactors:
- Moved this fan-out into `position.rs` alongside the other one mentioned above.